zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -556,7 +556,7 @@ impl Chain {
         // Don't add a new tree unless it differs from the previous one or there's no previous tree.
         if height.is_min()
             || self
-                .sprout_tree(height.previous().into())
+                .sprout_tree(height.previous().unwrap().into())
```

Comment:
It's OK to use `unwrap()` here because `previous()` gets executed only when `height.is_min()` returns `false` at line 557. Likewise for all the other `unwrap()`s in this PR.

Reply:
Usually in this situation we would say `expect("already checked for underflow")` so if we're wrong it's easier to diagnose the bug.

Reply:
Done in [bad9fbb](https://github.com/ZcashFoundation/zebra/pull/7591/commits/bad9fbbaf644248ba687a4a5f7da816c5abbfa57).